pwiz_tools/Skyline/Model/AuditLog/AuditLogEntry.cs

```diff
@@ -206,10 +207,31 @@ public static bool ReadFromFile(string fileName, out string loggedSkylineDocumen
             {
                 using (var alert = new AlertDlg())
                 {
-                    alert.Message = string.Format(
-                        AuditLogStrings.AuditLogList_ReadFromFile_An_exception_occured_while_reading_the_audit_log,
-                        fileName);
-                    alert.DetailMessage = ex.StackTrace;
+                    Exception exception = ex;
+                    StringBuilder msgBuilder = new StringBuilder("");
+                    List<string> stackList = new List<string>();
+
+
+                    do
+                    {
+                        if (msgBuilder.Length > 0)
+                        {
+                            msgBuilder.Append(@"--->");
+                            stackList.Add(Resources.ExceptionDialog_Caused_by_____);
+                        }
+                        msgBuilder.Append(exception.Message);
+                        stackList.Add(exception.Message);
+                        stackList.Add(exception.StackTrace);
+                        exception = exception.InnerException;
+                    } while (exception != null);
+
+                    alert.Message = TextUtil.LineSeparate(
+                        string.Format(
+                            AuditLogStrings.AuditLogList_ReadFromFile_An_exception_occured_while_reading_the_audit_log,
+                            fileName),
+                        msgBuilder.ToString());
+
+                    alert.DetailMessage = TextUtil.LineSeparate(stackList);
                     alert.AddMessageBoxButtons(MessageBoxButtons.OK);
                     alert.ShowParentlessDialog();
```

Comment:
This all looks pretty strange and contorted because of its placement inside Model. That is why the code ends up constructing its own AlertDlg and calling ShowParentlessDialog() (both really cheats to deal with the fact that we don't know what window the user is interacting with - and as Nick points out, there may be no window, if this is happening at the command-line). If the message code were properly placed, then the code to show a message with an OK button and an exception would be a 1-liner: MessageDlg.Show(parent, message, exception)

Reply:
The recent commits should take care of this issue.